sdk/appcenter/src/test/java/com/microsoft/appcenter/AppCenterTest.java

```diff
@@ -1117,4 +1121,46 @@ public void useApplicationLifecycleListener() {
         lifecycleListener.onActivityStopped(mockActivity);
         verify(service).onApplicationEnterBackground();
     }
+
+    @Test
+    public void setNetworkRequestValue() {
+        AppCenter.getInstance().setChannel(mChannel);
+
+        /* Disallow network request value. */
+        AppCenter.setNetworkRequestsAllowed(false);
+
+        /* Verify that value was passed to channel. */
+        verify(mChannel).setNetworkRequests(false);
+
+        /* Allow network request value. */
+        AppCenter.setNetworkRequestsAllowed(true);
+
+        /* Verify that value was passed to channel. */
+        verify(mChannel).setNetworkRequests(true);
+    }
+
+    @Test
+    public void setSameNetworkRequestsAllowedValue() {
+
+        /* Verify that default value is true. */
+        assertTrue(AppCenter.isNetworkRequestsAllowed());
+
+        /* Allow network request again. */
+        AppCenter.setNetworkRequestsAllowed(true);
+
+        /* Verify that value wasn't saved. */
+        verifyStatic(never());
+        SharedPreferencesManager.putBoolean(eq(PrefStorageConstants.ALLOWED_NETWORK_REQUEST), anyBoolean());
+
+        /* Disallow network requests. */
+        AppCenter.setNetworkRequestsAllowed(false);
+        assertFalse(AppCenter.isNetworkRequestsAllowed());
+        verifyStatic();
+        SharedPreferencesManager.putBoolean(eq(PrefStorageConstants.ALLOWED_NETWORK_REQUEST), anyBoolean());
+
+        /* Disallow network again requests. */
+        AppCenter.setNetworkRequestsAllowed(false);
+        verifyStatic();
+        SharedPreferencesManager.putBoolean(eq(PrefStorageConstants.ALLOWED_NETWORK_REQUEST), anyBoolean());
+    }
```

Comment:
Please add a test that service state doesn't affect channel/ingestion requests denial